src/plugin.ts

```diff
@@ -1,47 +1,70 @@
 import { Octokit } from "@octokit/rest";
-import { createClient } from "@supabase/supabase-js";
-import { createAdapters } from "./adapters";
-import { Env, PluginInputs } from "./types";
+import { Env, PluginInputs, SupportedEventsU } from "./types";
 import { Context } from "./types";
+import { askQuestion } from "./handlers/ask-gpt";
+import { addCommentToIssue } from "./handlers/add-comment";
+import { Logs } from "@ubiquity-dao/ubiquibot-logger";
 
 /**
  * How a worker executes the plugin.
  */
 export async function plugin(inputs: PluginInputs, env: Env) {
   const octokit = new Octokit({ auth: inputs.authToken });
-  const supabase = createClient(env.SUPABASE_URL, env.SUPABASE_KEY);
 
   const context: Context = {
     eventName: inputs.eventName,
     payload: inputs.eventPayload,
     config: inputs.settings,
     octokit,
     env,
-    logger: {
-      debug(message: unknown, ...optionalParams: unknown[]) {
-        console.debug(message, ...optionalParams);
-      },
-      info(message: unknown, ...optionalParams: unknown[]) {
-        console.log(message, ...optionalParams);
-      },
-      warn(message: unknown, ...optionalParams: unknown[]) {
-        console.warn(message, ...optionalParams);
-      },
-      error(message: unknown, ...optionalParams: unknown[]) {
-        console.error(message, ...optionalParams);
-      },
-      fatal(message: unknown, ...optionalParams: unknown[]) {
-        console.error(message, ...optionalParams);
-      },
-    },
-    adapters: {} as ReturnType<typeof createAdapters>,
+    logger: new Logs("info"),
+    adapters: {} as never,
   };
+  const {
+    logger,
+    config: { isEnabled },
+  } = context;
 
-  context.adapters = createAdapters(supabase, context);
+  if (isSupportedEvent(context.eventName)) {
+    const comment = context.payload.comment.body;
 
-  if (context.eventName === "issue_comment.created") {
-    // do something
+    if (!comment.startsWith("/gpt")) {
+      logger.info("Comment does not start with /gpt. Skipping.");
```

Comment:
Will this be a problem? It is not clear to me if the kernel will invoke this on any new comment.

Reply:
Truthfully me either.

The kernel does match using the regex provided in the plugin config so as far as I'm aware this probably isn't required it just felt like a good to have, I'll confirm and remove if not needed.

Reply:
@gentlementlegen might know.

Reply:
> ones like start stop which is only working for one of the two commands.

Wasn't it the case originally that the kernel was matching the command regex as opposed to `startsWith`? With regex `/(start|stop)` is technically one command so it shouldn't be treated like two as far as the kernel is concerned but the plugin would obviously have to split it.

> TL;DR it is not required but definitely safer to have the checks there.

I confirmed locally for me that it does fire on each comment so I'd say it is required or this plugin at least would comment an error for various reasons

Reply:
Related PR for the fix is [here](https://github.com/ubiquity/ubiquibot-kernel/issues/75).